### PR TITLE
New Board Selector UI: show port protocol

### PR DESCRIPTION
**Pull request by AlbyIanna**
### Motivation
This PR addresses a number of UX/UI improvements about the **Board Selector**.

#### This is the new **expected behavior** when starting with a sketch with no prior board selected:
1. the Board Selector shows "Select Board"
2. hovering over the Board Selector will make a tooltip appear saying "Disconnected"
3. clicking on the Board Selector will open the dropdown, showing a list of boards, with an icon on the left indicating the protocol (e.g.: usb)
4. clicking on a recognized board (not "Unknown") will close the dropdown and select the board
5. the Board Selector now shows the protocol icon on the left, followed by the name of the selected board
6. hovering over the Board Selector will make a tooltip appear showing the name off the board (e.g.: "Arduino MKR WiFi 1010") with a bold font
7. unplugging the board will make the protocol icon disappear and the board name font is not bold anymore
8. hovering over the Board Selector will make a tooltip appear saying "Disconnected"
9. replugging the board will make the protocol icon reappear and the font is bold again (as in step 5.)
10. clicking on the Board Selector and selecting an Unknown board will make the Select Board dialog appear with an overlay
11. Clicking "Cancel" will make the Select Board dialog diappear, and the Board Selector will show the label "Select Board" again
12. hovering over the Board Selector will make a tooltip appear saying "Disconnected"

### Change description
- Show a nice icon indicating which protocol the ports are using (`"serial"`, `"network"`, ecc...)
- Remove misguiding red cross indicating a board is disconnected
- Show the board/port info in a prettier way
- make boards in the dropdown focusable
- other cosmetic improvements 💄 

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull 
Requests](https://github.com/arduino/arduino-ide/pulls) before creating 
one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/src/common/protocol/boards-service.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ import { ArduinoComponent } from './arduino-component';
 
 export type AvailablePorts = Record<string, [Port, Array<Board>]>;
 export namespace AvailablePorts {
-  export function byProtocol(availablePorts: AvailablePorts): Map<string, AvailablePorts> {
+  export function byProtocol(
+    availablePorts: AvailablePorts
+  ): Map<string, AvailablePorts> {
```
Comment (AlbyIanna) [inline arduino-ide-extension/src/common/protocol/boards-service.ts:10]: On this file there's no significant change, I've just formatted the code 💄

**Comment by kittaakos** [comment]
It looks great! I did not check the code but the functionality this time.

It does not work correctly with the HC theme:
Screen Shot 2022-07-15 at 10 31 10

There is some overlay issue with the HC theme:

https://user-images.githubusercontent.com/1405703/179187194-544e2a9a-2d67-4a1b-83a5-1233e6eaba7b.mp4

No hover with HC or dark theme. It works with the light theme.

https://user-images.githubusercontent.com/1405703/179187307-bd484ec1-ea23-427f-8d5d-281e151a9781.mp4

I can unselect a board. This is a regression. I do not know if it's on purpose.

RC-8:

https://user-images.githubusercontent.com/1405703/179187418-8535047a-6a23-43af-8c31-f21ce9056ba5.mp4

From the PR:

https://user-images.githubusercontent.com/1405703/179187453-59309afe-69e2-403a-9c16-b0e36b1a3bc1.mp4

File: arduino-ide-extension/src/browser/boards/boards-service-provider.ts [hunk 2]
```diff
@@ -270,7 +270,6 @@ export class BoardsServiceProvider implements FrontendApplicationContribution {
   }
 
   protected setBoardsConfig(config: BoardsConfig.Config): void {
-    this.logger.info('Board config changed: ', JSON.stringify(config));
```
Comment (kittaakos) [inline arduino-ide-extension/src/browser/boards/boards-service-provider.ts:273]: Why did you remove it?

File: arduino-ide-extension/src/browser/boards/boards-toolbar-item.tsx [hunk 3]
```diff
@@ -28,10 +28,12 @@ export namespace BoardsDropDown {
 
 export class BoardsDropDown extends React.Component<BoardsDropDown.Props> {
   protected dropdownElement: HTMLElement;
+  listRef: React.RefObject<HTMLDivElement>;
```
Comment (kittaakos) [inline arduino-ide-extension/src/browser/boards/boards-toolbar-item.tsx:31]: missing `protected` or `private`?

File: arduino-ide-extension/src/browser/boards/boards-toolbar-item.tsx [hunk 4]
```diff
@@ -41,6 +43,14 @@ export class BoardsDropDown extends React.Component<BoardsDropDown.Props> {
     }
   }
 
+  override componentDidUpdate(prevProps: BoardsDropDown.Props): void {
+    if (prevProps.coords === 'hidden' && this.listRef.current) {
+      this.listRef.current.focus();
+    }
+  }
+
+  override componentDidMount(): void {}
```
Comment (kittaakos) [inline arduino-ide-extension/src/browser/boards/boards-toolbar-item.tsx:52]: Leftover?

File: arduino-ide-extension/src/browser/style/fonts.css [hunk 5]
```diff
@@ -713,4 +653,4 @@
 }
 .fa-microchip:before {
   content: "\f2db";
-}
+}
```
Comment (kittaakos) [inline arduino-ide-extension/src/browser/style/fonts.css:656]: Missing newline?

**Comment by AlbyIanna** [comment]
@kittaakos thank you for the functional review. @91volt is working to find the right color variables to fix dark and hc themes 🔨

File: arduino-ide-extension/src/browser/boards/boards-service-provider.ts (hunk 2 above)
Comment (AlbyIanna) [reply arduino-ide-extension/src/browser/boards/boards-service-provider.ts:273]: I still think it's kinda noisy during development, but you find it useful I'll re-add it, no prob 👍

**Comment by kittaakos** [comment]
Very nice! Is there a chance to restore the border of the board select dropdown on the HC theme? I assume it has nothing to do with the colors.

RC-8:
Screen Shot 2022-07-15 at 15 59 49


PR build:
Screen Shot 2022-07-15 at 15 59 11

**Comment by kittaakos** [comment]
> No hover with HC or dark theme. It works with the light theme.

Please open a follow-up once this is merged. Thank you!

**Comment by kittaakos** [comment]
Can you please remove this thing? It happens when I hover over the _board select_ dropdown. Please reference the screencast.

https://user-images.githubusercontent.com/1405703/179246924-516f5f74-e638-49c6-8dc5-d6f97de891a8.mp4

As I see it's always there when the cursor is above the toolbar:

https://user-images.githubusercontent.com/1405703/179247230-de2438b6-73c7-4f3f-ab6a-9615dad0b1ab.mp4

**Comment by AlbyIanna** [comment]
> Can you please remove this thing? It happens when I hover over the board select dropdown. Please reference the screencast.

It's weird because I saw that rectangle before, but I cannot reproduce it with the last commit 🤔 Did you notice you did something specific to "activate" this behavior?

**Review by kittaakos — APPROVED** [review]
I like the changes and the code looks OK. 👍 Thank you!

Somobody else who knows the requirements should also check the PR.

If you decide not to fix all review issues, please open a follow-up task with the TODOs.

**Comment by AlbyIanna** [comment]
Thanks @kittaakos!
> Somobody else who knows the requirements should also check the PR.

@91volt did a review today and he was okay with the changes but forgot to approve the PR.

> If you decide not to fix all review issues, please open a follow-up task with the TODOs.

We'll open an issue for the HC issues.

> Can you please remove this thing? It happens when I hover over the board select dropdown. Please reference the screencast.

About this "ghost rectangle" I'll open an issue when I manage to reproduce it again.
